feat(nuxt): reset asyncData to default value when calling clear/clearNuxtData (#27295)
nuxt/nuxt: docs/1.getting-started/12.upgrade.md

Please report an issue if you are doing this, as we do not plan to keep this as configurable.

> ### `clear` and `clearNuxtData` reset `data` in `useAsyncData` and `useFetch`

Author: ```suggestion
#### Respect defaults when clearing `data` in `useAsyncData` and `useFetch`
```

Please report an issue if you are doing this, as we do not plan to keep this as configurable.

### `clear` and `clearNuxtData` reset `data` in `useAsyncData` and `useFetch`

🚦 **Impact Level**: Minimal

##### What Changed

> If you provide a custom `default` value for `useAsyncData`, this will now be used when calling `clear` or `clearNuxtData` and it will bereset to its default value rather than simply unset.

Author: ```suggestion
If you provide a custom `default` value for `useAsyncData`, this will now be used when calling `clear` or `clearNuxtData` and it will be reset to its default value rather than simply unset.
```

##### Reasons for Change

Often users set an appropriately empty value, such as an empty array, to avoid the need to check for `null`/`undefined` when iterating over it. This should be respected when resetting/clearing the data.

##### Migration Steps

If you encounter any issues you can revert back to the previous behavior, for now, with:

```ts twoslash [nuxt.config.ts]
export default defineNuxtConfig({
  experimental: {
    resetAsyncDataToUndefined: true,
  }
})
```

Please report an issue if you are doing so, as we do not plan to keep this as configurable.

> 

Author: ```suggestion
```
